### PR TITLE
RPNV1-1097  Universe for large number of nodes

### PR DESCRIPTION
**Pull request by rvelaz**
Compress universe secret and store as AWS binary secret

### CODE REVIEW

File: radixdlt-core/radixdlt/src/main/java/org/radix/GenerateUniverses.java [hunk 1]
```diff
@@ -339,7 +343,7 @@ private static void outputNumberedKeys(
 		int n = 0;
 		List<Map<String, Object>> validators = new ArrayList<>();
 		final String VALIDATOR_PREFIX = "VALIDATOR";
-
+		Boolean compress = false;
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/org/radix/GenerateUniverses.java:346]: use plain `boolean` type

File: radixdlt-core/radixdlt/src/main/java/org/radix/GenerateUniverses.java [hunk 2]
```diff
@@ -405,8 +409,9 @@ private static void outputUniverse(
 					String filename = String.format("%s/universe.yaml", helmUniverseOutput.getHelmValuesPath());
 					writeYamlOutput(filename, config);
 				} else if (awsSecretsUniverseOutput.getEnableAwsSecrets()) {
+					Boolean compress = true;
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/org/radix/GenerateUniverses.java:412]: use plain `boolean` type

File: radixdlt-core/radixdlt/src/main/java/org/radix/GenerateUniverses.java [hunk 3]
```diff
@@ -450,11 +455,11 @@ private static void writeYamlOutput(String fileName, Map<String, Map<String, Obj
 		}
 	}
 
-	public static void writeAWSSecret(Map<String, Object> awsSecret, String secretName, AWSSecretsUniverseOutput awsSecretsUniverseOutput) {
+	public static void writeAWSSecret(Map<String, Object> awsSecret, String secretName, AWSSecretsUniverseOutput awsSecretsUniverseOutput, Boolean compress) {
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/org/radix/GenerateUniverses.java:458]: use plain `boolean` type

File: radixdlt-core/radixdlt/src/main/java/org/radix/GenerateUniverses.java [hunk 4]
```diff
@@ -490,7 +495,8 @@ private static void generateAWSSecrets(
 		final List<Map<String, Object>> validators,
 		final String validatorPrefix,
 		final String template,
-		final AWSSecretsUniverseOutput awsSecretsUniverseOutput
+		final AWSSecretsUniverseOutput awsSecretsUniverseOutput,
+		Boolean compress
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/org/radix/GenerateUniverses.java:499]: use plain `boolean` type

File: radixdlt-core/radixdlt/src/main/java/org/radix/universe/output/AWSSecretManager.java [hunk 5]
```diff
@@ -5,30 +5,40 @@
...
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
+import java.util.zip.GZIPOutputStream;
 
 public class AWSSecretManager {
     private static Region defaultRegion = Region.EU_WEST_2;
-    public static void createSecret(String secretName, String secretValue, String network, Region region) {
+
+    public static void createSecret(String secretName, Object secretValue, String network, Region region, Boolean binarySecret) {
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/org/radix/universe/output/AWSSecretManager.java:24]: use plain `boolean` type

File: radixdlt-core/radixdlt/src/main/java/org/radix/universe/output/AWSSecretManager.java [hunk 6]
```diff
@@ -90,49 +122,86 @@ private static String getValue(SecretsManagerClient secretsClient, String secret
 
     }
 
-    private static String createNewSecret(SecretsManagerClient secretsClient, String secretName, String secretValue, String network) {
+    private static String createNewSecret(SecretsManagerClient secretsClient, String secretName, Object secretValue, String network, Boolean binarySecret) {
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/org/radix/universe/output/AWSSecretManager.java:125]: use plain `boolean` type

**Review by Theoklitos — APPROVED** [review]

File: radixdlt-core/radixdlt/src/main/java/org/radix/universe/output/AWSSecretManager.java [hunk 7]
```diff
@@ -90,49 +122,86 @@ private static String getValue(SecretsManagerClient secretsClient, String secret
...
         List<Tag> tagList = buildTags(network, secretName);
-
-        CreateSecretRequest secretRequest = CreateSecretRequest.builder()
+		CreateSecretRequest secretRequest;
+        if (Boolean.TRUE.equals(binarySecret)){
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/org/radix/universe/output/AWSSecretManager.java:128]: With plain `boolean` type you can just use `if(binarySecret)`.

File: radixdlt-core/radixdlt/src/main/java/org/radix/universe/output/AWSSecretManager.java [hunk 8]
```diff
@@ -90,49 +122,86 @@ private static String getValue(SecretsManagerClient secretsClient, String secret
...
         CreateSecretResponse secretResponse = secretsClient.createSecret(secretRequest);
         return secretResponse.arn();
     }
+
     public static void createAWSSecret(
         final Map<String, Object> awsSecret,
         final String secretName,
-        final AWSSecretsUniverseOutput awsSecretsUniverseOutput
+        final AWSSecretsUniverseOutput awsSecretsUniverseOutput,
+        Boolean compress
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/org/radix/universe/output/AWSSecretManager.java:153]: use plain `boolean` type

File: radixdlt-core/radixdlt/src/main/java/org/radix/universe/output/AWSSecretManager.java [hunk 9]
```diff
@@ -90,49 +122,86 @@ private static String getValue(SecretsManagerClient secretsClient, String secret
...
     ) {
         ObjectMapper objectMapper = new ObjectMapper();
         try {
             String jsonSecret = objectMapper.writeValueAsString(awsSecret);
-            AWSSecretManager.createSecret(secretName, jsonSecret, awsSecretsUniverseOutput.getNetworkName());
+            if (Boolean.TRUE.equals(compress)){
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/org/radix/universe/output/AWSSecretManager.java:158]: With plain `boolean` type you can just use `if(compress)`.

File: radixdlt-core/radixdlt/src/main/java/org/radix/universe/output/AWSSecretManager.java [hunk 10]
```diff
@@ -90,49 +122,86 @@ private static String getValue(SecretsManagerClient secretsClient, String secret
...
+        GZIPOutputStream gzip = new GZIPOutputStream(bos);
+        gzip.write(data.getBytes());
+        gzip.close();
+        byte[] compressed = bos.toByteArray();
+        bos.close();
+        return compressed;
+    }
 
-    public static void updateAWSSecret(Map<String, Object> awsSecret, String secretName, AWSSecretsUniverseOutput awsSecretsUniverseOutput) {
+    public static void updateAWSSecret(Map<String, Object> awsSecret, String secretName, AWSSecretsUniverseOutput awsSecretsUniverseOutput, Boolean compress) {
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/org/radix/universe/output/AWSSecretManager.java:185]: Use plain `boolean` type

File: radixdlt-core/radixdlt/src/main/java/org/radix/universe/output/AWSSecretManager.java [hunk 11]
```diff
@@ -90,49 +122,86 @@ private static String getValue(SecretsManagerClient secretsClient, String secret
...
         ObjectMapper objectMapper = new ObjectMapper();
         if (canBeUpdated(awsSecretsUniverseOutput)) {
             System.out.format("Secret %s exists. And it's going to be replaced %n", secretName);
             try {
                 String jsonSecret = objectMapper.writeValueAsString(awsSecret);
-                updateSecret(secretName, jsonSecret);
+                if (Boolean.TRUE.equals(compress)){
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/org/radix/universe/output/AWSSecretManager.java:191]: With plain `boolean` type you can just use `if(binarySecret)`.

**Review by siy — APPROVED** [review]
LGTM with minor comments

**Pull request merged**